netsocadmin/register_tools.py

```diff
@@ -257,6 +300,74 @@ def remove_ldap_user(user: str) -> bool:
     with ldap3.Connection(ldap_server, auto_bind=True, receive_timeout=5, **config.LDAP_AUTH) as conn:
         return conn.delete(f"cn={user},cn=member,dc=netsoc,dc=co")
 
+def reset_password(user: str, email: str):
+    """
+    Sends an email once a user has reset their password
+
+    :param email the email address which this email is being sent
+    :param user the username which you log into the servers with
+    :returns True if the password was reset succesfully, False otherwise
+    """
+
+    with ldap3.Connection(ldap_server, auto_bind=True, receive_timeout=5, **config.LDAP_AUTH) as conn:
+        success = conn.search(
+            search_base="dc=netsoc,dc=co",
+            search_filter=f"(&(objectClass=account)(uid={user}))",
+            attributes=["uid", "gidNumber", "userPassword", "cn", "sn"],
+        )
+        if not success:
+            return False
+        # return conn.entries[0]["sn"]
+        entry = conn.entries[0]
+        old_password = entry["userPassword"]
+        password = "".join(random.choice(string.ascii_letters + string.digits) for _ in range(12))
+        # pylint: disable=E1101
+        crypt_password = "{crypt}" + crypt.crypt(password,  crypt.mksalt(crypt.METHOD_SHA512))
+        if not conn.modify(f"cn={user},cn=member,dc=netsoc,dc=co", {"userPassword": [(ldap3.MODIFY_REPLACE, [f"{crypt_password}"])]}):
+            conn.modify(f"cn={user},cn=admins,dc=netsoc,dc=co", {"userPassword": [(ldap3.MODIFY_REPLACE, [f"{crypt_password}"])]})
```

Comment:
Instead of trying one and then the other, you could just use an if else with the gidNumber, as gid=420 = admins and gid=422 = members